### PR TITLE
`BYOIP`: simplify configurations for other cluster-profiles to utilize

### PR DESCRIPTION
**Pull request by smg247**
Originally, this was hardcoded to only work on the `aws` cluster-profile. This configuration change will make it possible for others to be added as desired. I will be working on adding documentation for this in the cluster-profile area. There is an, in progress, change to cluster-profile configuration that we will be reworking these configs into once it lands.

Also fixed a flaky race condition with the, previously added, unit tests.

### CODE REVIEW

File: pkg/api/types.go [hunk 1]
```diff
@@ -1946,6 +1946,16 @@ func (p ClusterProfile) IPPoolLeaseType() string {
 	}
 }
 
+// IPPoolLeaseShouldValidateBranch declares whether the ip-pool leases should only be applied to branches matching a
+// specific OpenShift validation model. returns true by default, but should return false for any cluster-profiles
```
Comment (deepsm007) [inline pkg/api/types.go:1950]: Ideal condition would be to set it false by default and opt in? (based on only the cloud platforms are using it in this case for now aws)

Comment (smg247) [reply pkg/api/types.go:1950]: I thought about that, but I think the vast majority of use cases are going to want true. So I made it opt out.

**Pull request merged**